src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/TestRulesContext.cs

```diff
@@ -74,7 +76,7 @@ public T RegisterRule<T>() where T : ConsensusRule, new()
 
         public override RuleContext CreateRuleContext(ValidationContext validationContext, ChainedHeader tip)
         {
-            throw new System.NotImplementedException();
+           return this.RuleContext ?? new PowRuleContext();
```

Comment:
Not sure what all this code does but it's strange to return an already existing `Rulecontext` in a method called `CreateRuleContext` .

Reply:
Its the ability to plug in a test context in some cases.